R/age_model_class.R

```diff
@@ -0,0 +1,222 @@
+#' Defines an age-structured simple SEIR model.
+#' Slots of this class are as follows
+#'
+#' @slot name character representing name of model
+#' @slot output_names names of the compartments which are used by the
+#'     model.
+#' @slot parameter_names names of the initial conditions and parameters
+#'     which are used by the model.
+#' @slot parameters named list containing parameters of the model.
+#'     Initial values for each compartment, S0, E0, I0, R0.
+#'     Other parameters: a, b, c which represent the rates of changes
+#'     between the compartments.
+#' @slot n_age_categories number of age categories.
+#'
+#' @import deSolve
+#' @import glue
+#' @import ggplot2
+
+setClass('age_model',
+         #slots
+         slots = c(
+           name = 'character',
+           output_names = 'list',
+           parameter_names = 'list',
+           parameters = 'list',
+           n_age_categories = 'numeric'
+         ),
+
+         #prototypes for the slots, automatically set output and param
+         #names
+         prototype = list(
+           name = NA_character_,
+           output_names = list('S', 'E', 'I', 'R', 'Incidence'),
+           parameter_names = list('S0', 'E0', 'I0', 'R0', 'a', 'b', 'c'),
```

Comment:
currently we are using different names for the parameters. You guys are using a, b, and c and we (in the basic SEIR model) are using b(eta), k(appa), g(amma). It might be good for us to reach a consensus across the project on this. Happy to copy your parameter names if everyone is ok with that.

Reply:
Yep, I agree with going with the literature. Have changed ours to b, k and g.